src/bindings/pyrpr/src/pyrpr.py

```diff
@@ -561,14 +561,23 @@ def mark_static(self, is_static):
     def set_vertex_value(self, index: int, indices, values):
         ShapeSetVertexValue(self, index, ffi.cast("rpr_int *", indices.ctypes.data),
                             ffi.cast("float *", values.ctypes.data), len(indices))
-
+    
+    def set_primvar(self, key: int, data, floatcount: int, count: int, interop):
+        ShapeSetPrimvar(self, ffi.cast("rpr_uint",key), ffi.cast("float*", data.ctypes.data), 
+                        ffi.cast("rpr_uint",floatcount), ffi.cast("rpr_uint",count),
+                        ffi.cast("rpr_primvar_interpolation_type",interop))
+    
+    def set_vertex_attribute(self, key: int, data, floatcount: int, interop):
+        floatData = np.array(data)
+        floatData = floatData.flatten()
+        floatData = np.ascontiguousarray(floatData, dtype=np.float32)
+        self.set_primvar(key,floatData,  len(floatData), floatcount, interop)
+        
     def set_vertex_colors(self, colors):
-        indices = np.arange(len(colors), dtype=np.int32)
-
-        # index is 0-3 index (use for r,g,b,a)
-        for i in range(4):
-            values = np.ascontiguousarray(colors[:, i], dtype=np.float32)
-            self.set_vertex_value(i, indices, values)
+        vColors = np.array(colors)
+        vColors = vColors.flatten()
+        vColors = np.ascontiguousarray(vColors, dtype=np.float32)
+        self.set_primvar(99,vColors,  len(vColors), 4, PRIMVAR_INTERPOLATION_VERTEX)
```

Comment:
Why is the key here "99" ? I'm assuming this is a test.  You could make a dictionary of primvar names -> number

Reply:
I used 99 here based off of the sdk again. In terms of vertex color attributes, the name can be different between two meshes. 

eg, mesh1 vertex color's attribute might be named "Attribute" (default) and mesh2's might be named "color" 
there could even be a mesh3 which has two color attributes. one named "Attribute" and one named "color".

Assigning a static value to the active color attribute is the only way I can see to keep the active stuff active as the 'vertex color' attribute referenced elsewhere.

figure mesh1 would get 100, mesh 2 would get 101, and mesh 3 would have 100 and 102. How would the vertex color attribute know which to pick unless another list of what the active color attributes are per mesh was made.

I could be entirely misunderstanding, but that's how I interpreted it.

The sdk example started at 100, so so did I.